gas/config/tc-riscv.c

```diff
@@ -1377,9 +1377,9 @@ riscv_ip (char *str, struct riscv_cl_insn *ip, expressionS *imm_expr,
 		  if (imm_expr->X_op != O_constant
 		      || imm_expr->X_add_number <= 0
 		      || imm_expr->X_add_number >= RISCV_BIGIMM_REACH
-		      || (imm_expr->X_add_number >= RISCV_RVC_IMM_REACH / 2
```

Comment:
Here is some discussion about lui: https://github.com/riscv/riscv-binutils-gdb/issues/59